src/mappings/exchange.ts

```diff
@@ -32,8 +36,18 @@ function exponentToBigDecimal(decimals: i32): BigDecimal {
   return bd
 }
 
+function equalToZero(value: BigDecimal): boolean {
```

Comment:
If i remember correctly the reason I added this weird thing with float parsing is because the comparison wasn't working with BigDecimal.equals (which was giving us the divide by 0 error in the older version) - is there  fast way to test this? 

I'll do a slow test rigth now deploying on another subgraph and seeing if we get the error

Reply:
Hmm that could be possible. I am looking at the equals function and it doesn't appear to do any division:
```
  @operator('==')
  equals(other: ByteArray): boolean {
    if (this.length != other.length) {
      return false
    }
    for (let i = 0; i < this.length; i++) {
      if (this[i] != other[i]) {
        return false
      }
    }
    return true
  }
```

However, if you have it working with the function you have, then we can just skip it. I see no issue with this. It was just for code clean up

Reply:
yeah so that division was actually in the handleRemove liquidity - the images i sent were probably confusing. But basically they show that the value '0.000000000000' got passed into the new equalToZero function and that function returned false. 

So then it continued to execute this line 

```exchangeDayData.marginalEthRate = exchange.tokenBalance.div(exchange.ethBalance).truncate(8)```

even thought that ethBalnce was 0.

Reply:
basically the function wasn't evaluating to true when that 0.00000000000 ethBalance was passed in

Reply:
Ohhhhh! Gotcha. Funny enough I now remember we talked about this earlier last week within the company, we implemented a fix and it now should work for graph-ts 0.16.0 

It might be worth it to bump that to 0.16.0 and also graph-cli 0.16.0 as well. I don't think anything should break if we do it. But yeah you can do that after saturday when you guys release

Reply:
Awesome - I think since we're launching very soon I'll just open an issue for this and make a separate PR after this. Can test it then.